mesonbuild/compilers/compilers.py

```diff
@@ -1277,8 +1277,8 @@ def gen_vs_module_defs_args(self, defsfile):
         # For other targets, discard the .def file.
         return []
 
-    def get_gui_app_args(self):
-        if self.gcc_type in (GCC_CYGWIN, GCC_MINGW):
+    def get_gui_app_args(self, value):
+        if self.gcc_type in (GCC_CYGWIN, GCC_MINGW) and value:
             return ['-mwindows']
```

Comment:
Maybe this should mirror MSVC and explicitly pass `-mconsole` in the `else:` case?

Reply:
> See also: #3986 

That's kind of orthogonal, as it's about the behaviour when `gui_app:` is true